### ISSUE TITLE
Support GCS for filestore

### ISSUE THREAD

**Issue by jody-frankowski** [getsentry/sentry#7780]
It would be nice if Sentry could store its [files](https://docs.sentry.io/server/filestore/) in GCS natively.
For now, one can use the [interoperability mode](https://cloud.google.com/storage/docs/interoperability) but it is a bit more involved.

**Comment by mattrobenolt** [comment]
See: https://github.com/getsentry/sentry/pull/6133

**Comment by untoldone** [comment]
Does this work today? Given documentation, I tried storing something to GCS and nothing made it / but also no errors or logs in Sentry 9.0.0 beyond the webserver accepting it with a 201. I can tell its been configured correctly as later when I try to fetch the release file, I get an internal error with a 404 which shows the correct bucket name. I've also tested the credentials work separately from sentry.

Also, maybe needs another bug, but the docker image doesn't install the `google-cloud` dep for this
